src/data/tree.lean

```diff
@@ -136,9 +139,14 @@ by simp [unode_def]
 @[simp] lemma unode_ne_nil (a b : tree unit) : a.unode b ≠ nil := by trivial
 @[simp] lemma nil_ne_unode (a b : tree unit) : nil ≠ a.unode b := by trivial
 
-/-- Recursion on `tree unit` which uses `unode` -/
+/-- Recursion on `tree unit` using `unode` -/
+@[elab_as_eliminator]
 def unit_rec_on {motive : tree unit → Sort*} (t : tree unit) (base : motive nil)
   (ind : ∀ x y, motive x → motive y → motive (x.unode y)) : motive t :=
 t.rec_on base (λ u, u.rec_on (by exact ind))
 
+lemma left_unode_right_eq_self : ∀ {x : tree unit} (hx : x ≠ nil), x.left.unode x.right = x
+| nil h := by trivial
```

Comment:
```suggestion
```
Lean should be smart enough for you not to write this down.

Reply:
Simply deleting that line does not seem to work.